### ISSUE TITLE
UnsupportedOperationException: Unknown value type: 248

### ISSUE THREAD

**Issue by avcad** [jwhitbeck/java-rdb-parser#15]
Hello,

I am trying the parser and I  am seeing "Unknown value type: 248".  I wonder if you have any suggestions.
The issue seems to happen when I set an expire for the HSET. I created minimal code to reproduce the issue:

```
127.0.0.1:6379> INFO SERVER
# Server
redis_version:6.0.10
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:d4a632e656c1d2c6
redis_mode:standalone
```
Create a HSET with expire:
```
127.0.0.1:6379> FLUSHDB
OK
127.0.0.1:6379> HSET myhash values xxxx
(integer) 1
127.0.0.1:6379> EXPIRE myhash 900000
(integer) 1
127.0.0.1:6379> SAVE
OK
```
Error:
```
Exception in thread "main" java.lang.UnsupportedOperationException: Unknown value type: 248
	at net.whitbeck.rdbparser.RdbParser.readEntry(RdbParser.java:402)
	at net.whitbeck.rdbparser.RdbParser.readEntryMillis(RdbParser.java:366)
	at net.whitbeck.rdbparser.RdbParser.readNext(RdbParser.java:196)
	at debug.Debug.printRdbFile(Debug.java:15)
	at debug.Debug.main(Debug.java:9)
```
Remove the expire:
```
127.0.0.1:6379> PERSIST myhash
(integer) 1
127.0.0.1:6379> SAVE
OK
```
Now it works:
```
Processing DB: 0
------------
Key value pair
Key: myhash
Value type: HASHMAP_AS_ZIPLIST
Values: values xxxx 
------------
End of file. Checksum: c0e7e68c33cf9434
------------
```

I'd appreciate any suggestions. thanks! (hopefully I am not doing something dumb)

**Comment by jwhitbeck** [comment]
Thanks for reporting this error. At first sight, this looks like a bug.

To set expectations, I probably won't be able to investigate for this deeply for the next couple of weeks. 

As a shot in the dark, do you get the same error with EXPIREAT?

**Comment by avcad** [comment]
Hi, Yes. The problem still happens with PEXPIREAT.

I also tested this, which renders the same error

127.0.0.1:6379> FLUSHDB
OK
127.0.0.1:6379> SET keyname keyvalue EX 9000
OK
127.0.0.1:6379> SAVE
OK

**Comment by avcad** [comment]
![dump_rdb](https://user-images.githubusercontent.com/58915774/116133932-4acda700-a69d-11eb-9073-3aab50ae1dd4.jpg)

I just ran a couple more tests with:

===========
127.0.0.1:6379> SET keyname keyvalue EX 9000
===========

I found that the code works with Redis v4 (RDB version 8) but fails with Redis v6 (RDB Version 9)

Thanks!

**Comment by avcad** [comment]
Hi @jwhitbeck, I was wondering if you had plans to add this support ( idle/expiry with Redis 6 )

**Comment by jwhitbeck** [comment]
Yes, I'm still planning to. I've just been swamped with other work lately.

**Comment by jwhitbeck** [comment]
@avcad I just opened PR #16 to fix this bug. Let me know if this fixes the issue for you, and then I'll merge it and make a new release.

**Comment by avcad** [comment]
@jwhitbeck  amazing! Thanks! I tried it and it is working fine.

Btw, on Windows I am forced to rename the Aux.java class as I get this error "Aux.java is an invalid name on this platform"
https://stackoverflow.com/questions/19743865/eclipse-invalid-name-on-this-platform

You might want to rename it. I used "AuxEntry"

Thanks again!

**Comment by avcad** [comment]
Hi @jwhitbeck, I was wondering when you were planning to merge the PR, as I would like to start using the new release.
Thanks.

**Comment by jwhitbeck** [comment]
Hi @avcad,  I just pushed two new releases: 

- 1.5.2 includes the fix on #16 
- 2.0.0 includes a few additional breaking changes, such as renaming Aux.java to AuxField.java for better Windows compatibility. 

Thanks for your help in surfacing and troubleshooting this bug.

**Comment by avcad** [comment]
Thanks!
